### ISSUE TITLE
Empty output file when there are no issues

### ISSUE THREAD

**Issue by gigaSproule** [apigee/apigeelint#622]
For some reason, since `2.79.0`, if there are no errors in the report, then the output to console shows `[]`, but the file does not contain any contents. I've tried replicating this via the node REPL, but I cannot, so there must be something going on that I'm missing.

To recreate:
1. Create the directory structure `apiproxy/proxies`
2. Create the file `default.xml` under `apiproxy/proxies` with the following content
    ```yaml
    <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
    <ProxyEndpoint name="default">
        <Description>Test</Description>
        <HTTPProxyConnection>
            <BasePath>/test</BasePath>
        </HTTPProxyConnection>
        <PreFlow name="PreFlow">
            <Request>
            </Request>
            <Response>
            </Response>
        </PreFlow>
        <Flows />
        <RouteRule name="default">
            <URL>https://example.com</URL>
        </RouteRule>
    </ProxyEndpoint>
    ```
3. Run apigee lint
    ```shell
    npx apigeelint@2.78.0 --formatter=codeclimate.js --path=apiproxy --profile=apigeex --write=codeclimate.json
    # Should see `[]` printed to console
    cat codeclimate.json
    # Should see `[]` in codeclimate.json
    npx apigeelint@2.79.0 --formatter=codeclimate.js --path=apiproxy --profile=apigeex --write=codeclimate.json
    # Should see `[]` printed to console
    cat codeclimate.json
    # Should see nothing in codeclimate.json
    ```
